core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -227,8 +233,9 @@ object RequestChannel extends Logging {
       }
 
       val requestQueueTimeMs = nanosToMs(requestDequeueTimeNanos - startTimeNanos)
-      val apiLocalTimeMs = nanosToMs(apiLocalCompleteTimeNanos - requestDequeueTimeNanos)
-      val apiRemoteTimeMs = nanosToMs(responseCompleteTimeNanos - apiLocalCompleteTimeNanos)
+      val callbackRequestTimeNanos = callbackRequestDequeTimeNanos.getOrElse(0L) - callbackRequestCompleteTimeNanos.getOrElse(0L)
```

Comment:
1. Isn't it backwards?  The complete time should be larger than deque time.
2.  If we do the following logic, then it should work regardless of whether we arrive here before setting callbackRequestCompleteTimeNanos or after setting callbackRequestCompleteTimeNanos.

```
val callbackRequestTimeNanos = callbackRequestCompleteTimeNanos.getOrElse(endTimeNanos) - callbackRequestDequeTimeNanos.getOrElse(endTimeNanos)
````

Reply:
ooops -- good call.